From aa3cc6e3b04e19166c8898e2d3d6918270b838b0 Mon Sep 17 00:00:00 2001
From: Jiawei Huang <jiawei@tigera.io>
Date: Wed, 10 Aug 2022 12:16:40 -0700
Subject: [PATCH 1/1] Remove password prompt in bin/elasticsearch script

For none interactive shell like Elasticsearch Pod startup in k8s
cluster, we don't want the user prompt for password encrypted Elastic
keystore. the password will be taken from environment variable
`KEYSTORE_PASSWORD`.
---
 bin/elasticsearch | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/bin/elasticsearch b/bin/elasticsearch
index 0e6243b..bbc87ac 100755
--- a/bin/elasticsearch
+++ b/bin/elasticsearch
@@ -37,19 +37,6 @@ if [ -z "$LIBFFI_TMPDIR" ]; then
   export LIBFFI_TMPDIR
 fi
 
-# get keystore password before setting java options to avoid
-# conflicting GC configurations for the keystore tools
-unset KEYSTORE_PASSWORD
-KEYSTORE_PASSWORD=
-if [[ $CHECK_KEYSTORE = true ]] \
-    && bin/elasticsearch-keystore has-passwd --silent
-then
-  if ! read -s -r -p "Elasticsearch keystore password: " KEYSTORE_PASSWORD ; then
-    echo "Failed to read keystore password on console" 1>&2
-    exit 1
-  fi
-fi
-
 # The JVM options parser produces the final JVM options to start Elasticsearch.
 # It does this by incorporating JVM options in the following way:
 #   - first, system JVM options are applied (these are hardcoded options in the
-- 
2.34.1

